File: from-2401-to-2500/2491-divide-players-into-teams-of-equal-skill/solution.go
func dividePlayers(skill []int) int64 {
    sort.Ints(skill)
    left, right := 0, len(skill) - 1
    es := skill[left] + skill[right]
    var result int64 = 0
    for left < right {
        if skill[left] + skill[right] != es {
            return -1
        }
        result += int64(skill[left] * skill[right])
        left += 1
        right -= 1
    }
    return result
}

File: from-2401-to-2500/2491-divide-players-into-teams-of-equal-skill/solution.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        teamSkill = skill[0] + skill[-1]
        l, r = 1, len(skill) - 2
        result = skill[0] * skill[-1]
        while l < r:
            if skill[l] + skill[r] != teamSkill:
                return -1
            result += skill[l] * skill[r]
            l += 1
            r -= 1
        return result